app/server/models/user.py

```diff
@@ -570,40 +571,20 @@ def get_most_relevant_transfer_usage(self):
         '''
 
         sql = text('''
-            SELECT *, COUNT(*)  FROM
-                (SELECT u.business_usage_id FROM credit_transfer c
-                LEFT JOIN "user" u ON c.recipient_user_id = u.id
-                WHERE sender_user_id = {} AND c.transfer_status = 'COMPLETE'
+            SELECT *, COUNT(*) FROM
+                (SELECT c.transfer_use::text FROM credit_transfer c
+                WHERE c.sender_user_id = {} AND c.transfer_status = 'COMPLETE'
                 ORDER BY c.updated DESC
                 LIMIT 20)
-            C GROUP BY business_usage_id ORDER BY count DESC
+            C GROUP BY transfer_use ORDER BY count DESC
         '''.format(self.id))
         result = db.session.execute(sql)
-        transfer_usage_ids = [row[0] for row in result]
-        # Get most common business_usage_id
-        number_of_wanted_business_id = 9
-
-        most_common_ids = transfer_usage_ids[:number_of_wanted_business_id]
-        # If less than the wanted nr of usage is found fill up the list with default
-        if len(most_common_ids) < number_of_wanted_business_id:
-            most_relevant_ids = self.refine_with_default_transfer_usages(
-                most_common_ids, number_of_wanted_business_id)
-        relevant_transer_usages = TransferUsage.query.filter(
-            TransferUsage.id.in_(most_common_ids)).all()
-        # Order the transfer usages by the ordered list of id that were ordered by count
-        ordered_tranfer_usages = [
-            next(s for s in relevant_transer_usages if s.id == id) for id in most_common_ids]
-        return ordered_tranfer_usages
-
-    def refine_with_default_transfer_usages(self, most_common, nr_of_wanted):
-        default_transer_usages = TransferUsage.query.filter_by(default=True).with_entities(
-            TransferUsage.id).all()
-        for transer_usage_id, in default_transer_usages:
-            if transer_usage_id not in most_common:
-                most_common.append(transer_usage_id)
-            if len(most_common) == nr_of_wanted:
-                break
-        return most_common
+        most_common_uses = {}
+        for row in result:
+            for use in json.loads(row[0]):
```

Comment:
it's a bit strange that transfer use is a json field that's an array of transfer usage names. harder to look up + perform sql queries on what's the reason for not making a join table of TransferUsage ids to CreditTransfers?